src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -0,0 +1,143 @@
+        rightPIDController = rightMotor1.getPIDController();
+        
+        leftEncoder = leftMotor1.getEncoder();
+        rightEncoder = rightMotor1.getEncoder();
+        
+        P = 0.00010; 
+        I = 0;
+        D = .0000; 
+        Iz = 0; 
+        FF = 0.000175; 
+        maxOutput = 1; 
+        minOutput = -1;
+
+        leftPIDController.setP(P);
+        leftPIDController.setI(I);
+        leftPIDController.setD(D);
+        leftPIDController.setIZone(Iz);
+        leftPIDController.setFF(FF);
+        leftPIDController.setOutputRange(minOutput, maxOutput);
+        
+        rightPIDController.setP(P);
+        rightPIDController.setI(I);
+        rightPIDController.setD(D);
+        rightPIDController.setIZone(Iz);
+        rightPIDController.setFF(FF);
+        rightPIDController.setOutputRange(minOutput, maxOutput);
+        
+        RobotContainer.sbTab.add("Left Drive Encoder", leftEncoder).withWidget(BuiltInWidgets.kEncoder).withPosition(0, 0).withSize(2, 1);
+        RobotContainer.sbTab.add("Right Drive Encoder", rightEncoder).withWidget(BuiltInWidgets.kEncoder).withPosition(2, 0).withSize(2, 1);
+        RobotContainer.sbTab.add("Left Drive PID", leftPIDController).withWidget(BuiltInWidgets.kPIDController).withPosition(0, 1).withSize(1, 2);
+        RobotContainer.sbTab.add("Right Drive PID", rightPIDController).withWidget(BuiltInWidgets.kPIDController).withPosition(4, 1).withSize(1, 2);
+        
+        SmartDashboard.putNumber("Drive - P", P);
+        SmartDashboard.putNumber("Drive - I", I);
+        SmartDashboard.putNumber("Drive - D", D);
+        SmartDashboard.putNumber("Drive - Iz", Iz);
+        SmartDashboard.putNumber("Drive - FF", FF);
+        SmartDashboard.putNumber("Drive - minOutput", minOutput);
+        SmartDashboard.putNumber("Drive - maxOutput", maxOutput);
+    }
+
+    public void updatePID()
+    {
+        double p = SmartDashboard.getNumber("Drive - P", 0);
+        double i = SmartDashboard.getNumber("Drive - I", 0);
+        double d = SmartDashboard.getNumber("Drive - D", 0);
+        double min = SmartDashboard.getNumber("Drive - minOutput", 0);
+        double max = SmartDashboard.getNumber("Drive - maxOutput", 0);
+        
+        if(p != P)
+        {
+            leftPIDController.setP(p);
+            rightPIDController.setP(p);
+            P = p;
+        }
+        if(i != I)
+        {
+            leftPIDController.setI(i);
+            rightPIDController.setI(i);
+            I = i;
+        }
+        if(d != D)
+        {
+            leftPIDController.setD(d);
+            rightPIDController.setD(d);
+            D = d;
+        }
+        if(max != maxOutput || min != minOutput)
+        {
+            minOutput = min;
+            maxOutput = max;
+        }
+    }
+    
+    public void arcadeDrive(double velocity, double heading)
+    {
+        differentialDrive.arcadeDrive(velocity, heading * 0.7, true);
```

Comment:
Remove the .7 * heading, move the .7 * heading to where you supply the joystick value for heading.

Reply:
You do this incase you want to make use of arcade drive in autonomous using non operator control input, you don't always want to multiple your heading command by .7